patches/server/0338-Optimise-EntityGetter-getPlayerByUUID.patch

```diff
@@ -1,23 +1,24 @@
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Spottedleaf <Spottedleaf@users.noreply.github.com>
 Date: Sat, 11 Jan 2020 21:50:56 -0800
-Subject: [PATCH] Optimise IEntityAccess#getPlayerByUUID
+Subject: [PATCH] Optimise EntityGetter#getPlayerByUUID
 
-Use the world entity map instead of iterating over all players
+Use the PlayerList map instead of iterating over all players
 
 diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
-index 0c5065ac62d8a708f70282e765277866834169bc..878123befd7580082cf85222fa3a331baf41156c 100644
+index 72d3af43afd1b71d92a4d0673795835c6a3a07d0..c38229f3844328e8e5a0d84e64a0867bafadc5dc 100644
 --- a/src/main/java/net/minecraft/server/level/ServerLevel.java
 +++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
-@@ -423,6 +423,14 @@ public class ServerLevel extends Level implements WorldGenLevel {
+@@ -423,6 +423,15 @@ public class ServerLevel extends Level implements WorldGenLevel {
      }
      // Paper end - rewrite chunk system
  
 +    // Paper start - optimise getPlayerByUUID
 +    @Nullable
 +    @Override
 +    public Player getPlayerByUUID(UUID uuid) {
-+        return this.getServer().getPlayerList().getPlayer(uuid);
++        final Player player = this.getServer().getPlayerList().getPlayer(uuid);
++        return player != null && player.level == this ? player : null;
```

Comment:
Don't you need parentheses here?  Because what you have is essentially:
`return (player != null && player.level == this) ? player : null;`
so you're comparing a boolean to a player and the return will always be null.
I think you want:
`return player != null && (player.level == this ? player : null);`
I didn't look to see if this is wrong elsewhere in the PR.